controller.py

```diff
@@ -0,0 +1,138 @@
+"""
+The main controller for our world. Initializes the world, camera, and renderer, and
+takes user input in order to control the camera's position in the world.
+"""
+
+
+from world import World
+from camera import Camera
+from renderer import Renderer
+from item import Item
+import pygame
+import threading
+import sys
+
+
+class Scene:
+    def __init__(self, window_size=(1000, 1000)):
+        """
+        Initializes a new scene.  By default, puts one object in and sets up everything in the correct positions.
+        """
+        self.window_size = window_size
+        self.world = World()
+        self.world.gen_random_scene(1, 3)
+        # self.world.add_item([Item('teapot.stl', (0, 0, 0), (0, 0, 0), 1, color=(255, 255, 0))])
+        # self.world.add_item([Item('cube.stl', (50, 10, 10), (0, 0, 0), 1, color=(255, 255, 0))])
+        self.camera = Camera(init_pos=[0, 1, -10], init_angle=[0, 0, 0], init_fov=1.57)
+        self.renderer = Renderer(self.camera, window_size)
+        self.running = False
+
+    def begin_scene(self):
+        """
+        Begins the rendering and user input MVC loop.
+        """
+        thread_lock = threading.Lock()  # lock needed to ensure threading is happy
+        canvas = pygame.display.set_mode(self.window_size, 0, 32)
+        pygame.mouse.set_visible(False)
+        # pygame.event.set_grab(True)  # Only uncomment this if you're SURE it won't break everything
+        render_thread = threading.Thread(target=self.render_cycle, args=(canvas, thread_lock,))
+        render_thread.start()
+
+        input_thread = threading.Thread(target=self.handle_user_input, args=(thread_lock,))
+        input_thread.start()
+
+    def render_cycle(self, canvas, lock):
+        """
+        handles the continuous cycle of rendering each frame.
+        """
+        pygame.init()
+        clock = pygame.time.Clock()
+        self.running = True
+        try:
+            while self.running:
+                lock.acquire()  # Acquire lock, otherwise things get weird
+                self.renderer.draw_scene(self.world, canvas)
+                lock.release()
+                clock.tick(60)  # FPS
+        except Exception as e:  # Avoid input grabbing issues by quitting on exception
```

Comment:
This is very dangerous things to do! This is ignoring all types of exceptions, including some of them that might be caused by other parts of your program that you want to see. Better way to do this is to figure out the types of error that you want to skip and specifying that type of exception.